src/coreclr/vm/instancecalli.cpp

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+
+void MethodDesc::GenerateFunctionPointerCall(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(resolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+    _ASSERTE(*resolver == NULL && *methodILDecoder != NULL);
```

Comment:
@AaronRobinsonMSFT can you take a look at these Asserts; these were copied from the unsafe accessor work but since there is IL in these methods `*methodILDecoder != NULL` and `GetRVA() != 0` I changed them to reflect that. Not sure if we need to worry about freeing any memory such as the existing `methodILDecoder`.

Reply:
We don't need to free anything here, it is all allocated on the stack. I would keep this assert as expecting `NULL` and set the `Header` field on `MethodInfoHelperContext` in the caller to `NULL`. Since this is an **identified** intrinsic it can be ignored. Adding a comment prior to calling this API to that effect would be appreciated.